### PR TITLE
plugins: add missing "description" metadata tags

### LINKED ISSUE #4383: plugins: add missing "description" metadata tags
Plugin metadata comments/tags were added via #4374 in favor of having a static plugin list/table in the docs (see #4369).
- https://streamlink.github.io/plugins.html
- https://streamlink.github.io/developing.html#plugins

The old plugin matrix didn't include any plugin descriptions due to space constraints of the table layout, and #4374 didn't add the missing metadata, because it would've delayed the implementation by a lot.

The `description` tag is therefore currently not marked as a requirement, but the goal is to have every existing plugin properly tagged with a short description text, so the tag can be made a requirement for successful builds eventually. New plugins should always include this tag and existing ones should get updated bit by bit.

As already mentioned in #4374, a couple of plugin descriptions can be found in the git history of the individual plugin files, because some old comments with descriptions were removed in the past for consistency reasons while refactoring code or while removing deprecated stuff. Other plugin descriptions can be found on the issue tracker in plugin request threads, but the rest will need to be looked up / researched.

There are about 150 built-in plugins, so if anyone feels like making a contribution by adding some of the missing metadata, please open a pull request. Thanks.

### PR DESCRIPTION
**Pull request by mkbloke**
closes #4383

### CODE REVIEW

**Comment by bastimeyer** [comment]
Thanks for putting in all the effort and researching this. I will review this properly later.

One mistake I noticed while quickly going through the list were the descriptions of ard_live and ard_mediathek at the top. It's the same TV channel, "ARD".

File: src/streamlink/plugins/mediavitrina.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 """
+$description Russian live TV channels from CTC Group, including Che, CTC, CTC Love and Domashny.
```
Comment (back-to) [inline src/streamlink/plugins/mediavitrina.py:2]: there is more than `CTC Group`, description should be like [cdnbg.py](https://github.com/streamlink/streamlink/pull/4391/files#diff-e7158d7c7de978893a186eacead94ed6f36f018eacf706dc19f91456c862a831)

File: src/streamlink/plugins/sportschau.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 """
+$description Sporting channel live stream owned by Sportschau, a German sports media website.
```
Comment (back-to) [inline src/streamlink/plugins/sportschau.py:2]: > Sporting channel

it is not a tv channel, just a sports magazine https://en.wikipedia.org/wiki/Sportschau

> owned by Sportschau

No, it is the same as ARD plugin

File: src/streamlink/plugins/stadium.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 """
+$description Sporting channel live stream and videos owned by Stadium, an American sports media website.
```
Comment (back-to) [inline src/streamlink/plugins/stadium.py:2]: > owned by Stadium

seems weird

> Stadium is a joint venture between [Silver Chalice](https://en.wikipedia.org/wiki/Silver_Chalice) and [Sinclair Broadcast Group](https://en.wikipedia.org/wiki/Sinclair_Broadcast_Group).

https://en.wikipedia.org/wiki/Stadium_(sports_network)

File: src/streamlink/plugins/teamliquid.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 """
+$description Live streams from Team Liquid, a multi-regional professional e-sports organisation based in the Netherlands.
```
Comment (back-to) [inline src/streamlink/plugins/teamliquid.py:2]: > Live streams from Team Liquid

it is not just them, there are more https://tl.net/video/streams/

but this plugin should be removed

File: src/streamlink/plugins/welt.py [hunk 5]
```diff
@@ -1,4 +1,5 @@
 """
+$description Two 24-hour channels covering news & features and video content from Die Welt, based in Germany.
```
Comment (back-to) [inline src/streamlink/plugins/welt.py:2]: > N24 Doku ... which is a one-hour delayed timeshift channel of Welt
https://en.wikipedia.org/wiki/Welt_(German_TV_channel)

wound't count it as two channels

> content from Die Welt

might be the same name, but the description is weird.

It is just a news channel with documentaries inbetween

File: src/streamlink/plugins/funimationnow.py [hunk 6]
```diff
@@ -1,4 +1,5 @@
 """
+$description Anime video on-demand service with over 800 titles.
```
Comment (back-to) [inline src/streamlink/plugins/funimationnow.py:2]: > with over 800 titles.

unimportant

File: src/streamlink/plugins/filmon.py [hunk 7]
```diff
@@ -1,4 +1,5 @@
 """
+$description Live local TV with over 600 channels and 90,000 video on-demand titles. OTT service from FilmOn.
```
Comment (back-to) [inline src/streamlink/plugins/filmon.py:2]: > 600 channels and 90,000

unimportant

File: src/streamlink/plugins/bfmtv.py [hunk 8]
```diff
@@ -1,4 +1,5 @@
 """
+$description BFM: National & regional news, weather and business. 01TV: technology news and topics. Both based in France.
```
Comment (back-to) [inline src/streamlink/plugins/bfmtv.py:2]: > BFM:
> 01TV:

don't split it, you could use less details

Comment (bastimeyer) [reply src/streamlink/plugins/bfmtv.py:2]: https://en.wikipedia.org/wiki/NextRadioTV

File: src/streamlink/plugins/mediaklikk.py [hunk 9]
```diff
@@ -1,4 +1,5 @@
 """
+$description Live TV channels from MTV, a Hungarian public, state-owned broadcaster.
```
Comment (bastimeyer) [inline src/streamlink/plugins/mediaklikk.py:2]:
```suggestion
$description Live TV channels from MTVA, a Hungarian public, state-owned broadcaster.
```

File: src/streamlink/plugins/nimotv.py [hunk 10]
```diff
@@ -1,4 +1,5 @@
 """
+$description Global live streaming platform for live video game broadcasts and individual live streams.
```
Comment (bastimeyer) [inline src/streamlink/plugins/nimotv.py:2]:
```suggestion
$description Chinese, global live streaming platform run by Huya Live.
```

File: src/streamlink/plugins/pluto.py [hunk 11]
```diff
@@ -1,4 +1,5 @@
 """
+$description Over 250 channels and 1,000s of video on-demand titles. OTT service from Pluto.
```
Comment (bastimeyer) [inline src/streamlink/plugins/pluto.py:2]:
```suggestion
$description Live TV and video on-demand service owned by Paramount Streaming.
```

File: src/streamlink/plugins/theplatform.py [hunk 12]
```diff
@@ -1,4 +1,5 @@
 """
+$description Global video on-demand hosting platform.
```
Comment (bastimeyer) [inline src/streamlink/plugins/theplatform.py:2]:
```suggestion
$description Global video on-demand hosting platform owned by Comcast.
```

File: src/streamlink/plugins/twitch.py [hunk 13]
```diff
@@ -1,4 +1,5 @@
 """
+$description Global live streaming and video hosting social platform.
```
Comment (bastimeyer) [inline src/streamlink/plugins/twitch.py:2]:
```suggestion
$description Global live streaming and video hosting social platform owned by Amazon.
```

File: src/streamlink/plugins/ustreamtv.py [hunk 14]
```diff
@@ -1,4 +1,5 @@
 """
+$description Global live streaming and video on-demand platform.
```
Comment (bastimeyer) [inline src/streamlink/plugins/ustreamtv.py:2]:
```suggestion
$description Global live streaming and video on-demand platform owned by IBM.
```

File: src/streamlink/plugins/welt.py (hunk 5 above)
Comment (bastimeyer) [inline src/streamlink/plugins/welt.py:2]:
```suggestion
$description German news and documentaries TV channel, owned by Axel Springer SE.
```

File: src/streamlink/plugins/youtube.py [hunk 15]
```diff
@@ -1,4 +1,5 @@
 """
+$description Global live streaming and video hosting social platform.
```
Comment (bastimeyer) [inline src/streamlink/plugins/youtube.py:2]:
```suggestion
$description Global live streaming and video hosting social platform owned by Google.
```

**Review by bastimeyer — COMMENTED** [review]
A couple of suggestions and corrections, in addition to what @back-to already reviewed.

File: src/streamlink/plugins/teamliquid.py (hunk 4 above)
Comment (mkbloke) [reply src/streamlink/plugins/teamliquid.py:2]: Let's leave it then - I'll submit a PR for removal of this plugin.

**Review by bastimeyer — COMMENTED** [review]
This seems to be good now. In case someone else wants to (re-)review, I'm leaving this PR open for another bit.

Thanks again for the work, @mkbloke. :+1: 

Btw, should we add another attribute to the metadata? Like for example what kind of streams the plugins return and whether FFmpeg is a requirement? This could probably be automated by checking the imports though.

**Review by bastimeyer — APPROVED** [review]

**Review by gravyboat — APPROVED** [review]
Looks good to me as well, thanks @mkbloke!

**Pull request merged**